geopandas/geodataframe.py

```diff
@@ -225,6 +250,39 @@ def rename_geometry(self, col, inplace=False):
         self.rename(columns={geometry_col: col}, inplace=inplace)
         self.set_geometry(col, inplace=inplace)
 
+    @property
+    def crs(self):
+        """
+        The Coordinate Reference System (CRS) represented as a ``pyproj.CRS``
+        object.
+
+        Returns None if the CRS is not set, and to set the value it
+        :getter: Returns a ``pyproj.CRS`` or None. When setting, the value
+        can be anything accepted by
+        :meth:`pyproj.CRS.from_user_input() <pyproj.crs.CRS.from_user_input>`,
+        such as an authority string (eg "EPSG:4326") or a WKT string.
+        """
+        return self._crs
+
+    @crs.setter
+    def crs(self, value):
+        """Sets the value of the crs"""
+        if self._geometry_column_name not in self:
+            warnings.warn(
+                "Assigning CRS to a GeoDataFrame without a geometry column is now "
+                "deprecated and will not be supported in the future.",
+                FutureWarning,
+                stacklevel=4,
+            )
+            self._crs = None if not value else CRS.from_user_input(value)
+        else:
+            if hasattr(self.geometry.values, "crs"):
+                self.geometry.values.crs = None if not value else value
+                self._crs = self.geometry.values.crs
+            else:
+                # column called 'geometry' without geometry
```

Comment:
Is it possible to end up here? As you already checked for `if self._geometry_column_name not in self:` above?